managed/resource_db_query_logging.go

```diff
@@ -0,0 +1,414 @@
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r resourceDbQueryLogging) Update(ctx context.Context, req tfsdk.UpdateResourceRequest, resp *tfsdk.UpdateResourceResponse) {
+	resp.Diagnostics.AddError(
+		"Unsupported Operation",
+		"Update is not currently supported.",
+	)
+}
+
+func (r resourceDbQueryLogging) Delete(ctx context.Context, req tfsdk.DeleteResourceRequest, resp *tfsdk.DeleteResourceResponse) {
+	resp.Diagnostics.AddError(
+		"Unsupported Operation",
+		"Delete is not currently supported.",
+	)
+}
+
+func (r resourceDbQueryLogging) ImportState(ctx context.Context, req tfsdk.ImportResourceStateRequest, resp *tfsdk.ImportResourceStateResponse) {
+	resp.Diagnostics.AddError("Import is not currently supported", "")
+
+}
+
+// Create a new Db Query Log Configuration for a Cluster
+func (r resourceDbQueryLogging) Create(ctx context.Context, req tfsdk.CreateResourceRequest, resp *tfsdk.CreateResourceResponse) {
+	if !r.p.configured {
+		resp.Diagnostics.AddError(
+			"Provider not configured",
+			"The provider wasn't configured before being applied, likely because it depends on an unknown value from another resource.",
+		)
+		return
+	}
+
+	var config DbQueryLoggingConfig
+	resp.Diagnostics.Append(getConfigFromPlan(ctx, req.Plan, &config)...)
+	if resp.Diagnostics.HasError() {
+		tflog.Debug(ctx, "Error while getting the plan for the DB query log configuration")
+		return
+	}
+
+	apiClient := r.p.client
+	var accountId, projectId string
+	accountId, ok, message := getAccountId(ctx, apiClient)
+	if !ok {
+		resp.Diagnostics.AddError("Unable to get account ID", message)
+		return
+	}
+
+	projectId, ok, message = getProjectId(ctx, apiClient, accountId)
+	if !ok {
+		resp.Diagnostics.AddError("Unable to get project ID", message)
+		return
+	}
+
+	var clusterId string
+	var integrationId string
+
+	clusterName := config.ClusterName.Value
+	integrationName := config.IntegrationName.Value
+	clusterData, err := GetClusterByNameorID(accountId, projectId, "", clusterName, apiClient)
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to get cluster id for cluster: "+clusterName, GetApiErrorDetails(err))
+		return
+	}
+	clusterId = clusterData.Info.Id
+
+	integrationConfig, _, err := apiClient.TelemetryProviderApi.
+		ListTelemetryProviders(context.Background(), accountId, projectId).
+		Name(integrationName).
+		Execute()
+
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to get integration ID for integration: "+integrationName, GetApiErrorDetails(err))
+		return
+	}
+
+	if len(integrationConfig.GetData()) < 1 {
+		message := fmt.Sprintf("Integration %s not found", integrationName)
+		resp.Diagnostics.AddError(message, message)
+		return
+	}
+	integrationId = integrationConfig.GetData()[0].Info.Id
+
+	dbQueryLoggingConfigSpec, err := buildDbQueryLoggingSpec(config, integrationId)
+	if err != nil {
+		resp.Diagnostics.AddError("Unable to build DB query logging config spec", GetApiErrorDetails(err))
```

Comment:
Looks like this error message is very specific to how our code works. I think a generalized error message like - 
"Encountered error while enabling DB Query Logging" should be shown to the user. Additional details can be added in the form of a Warning log or a debug log. Thoughts @bhupendray-yb

Reply:
Totally makes sense.